src/rust/engine/process_execution/src/remote.rs

```diff
@@ -384,6 +400,53 @@ mod tests {
     );
   }
 
+  #[test]
+  fn wait_between_request() {
+    // wait at least 500 milli for one retry
+    {
+      let execute_request = echo_foo_request();
+
+      let mock_server = {
+        let op_name = "gimme-foo".to_string();
+
+        mock::execution_server::TestServer::new(mock::execution_server::MockExecution::new(
+          op_name.clone(),
+          super::make_execute_request(&execute_request).unwrap().1,
+          vec![
+            make_incomplete_operation(&op_name),
+            make_successful_operation(&op_name, "foo", "", 0),
+          ],
+        ))
+      };
+      let start_time = SystemTime::now();
+      let result = run_command_remote(&mock_server.address(), execute_request).unwrap();
+      assert!(start_time.elapsed().unwrap() >= Duration::from_millis(500));
+    }
+
+    // wait at least 500 + 1000 + 1500 = 3000 milli for 3 retries.
+    {
+      let execute_request = echo_foo_request();
+
+      let mock_server = {
+        let op_name = "gimme-foo".to_string();
+
+        mock::execution_server::TestServer::new(mock::execution_server::MockExecution::new(
+          op_name.clone(),
+          super::make_execute_request(&execute_request).unwrap().1,
+          vec![
+            make_incomplete_operation(&op_name),
+            make_incomplete_operation(&op_name),
+            make_incomplete_operation(&op_name),
+            make_successful_operation(&op_name, "foo", "", 0),
+          ],
+        ))
+      };
+      let start_time = SystemTime::now();
+      let result = run_command_remote(&mock_server.address(), execute_request).unwrap();
+      assert!(start_time.elapsed().unwrap() >= Duration::from_millis(3000));
```

Comment:
It _could_ be nice to make the mock server record the time it gets requests, and assert on the gaps between them, rather than counting the total time. But I can also believe it's not worth the complexity :)